### PR TITLE
10/n Remove BeaconBlockT and BeaconStateT generic types

### PR DESCRIPTION
**Pull request by fridrik01**

### CODE REVIEW

**Review by nidhi-singh02 — APPROVED** [review]
looks good

File: cmd/beacond/types.go [hunk 1]
```diff
@@ -141,25 +131,15 @@ type (
 	AvailabilityStore = dastore.Store
 
 	// BeaconBlock type aliases.
-	ConsensusBlock = consruntimetypes.ConsensusBlock[*BeaconBlock]
-	BeaconBlock    = types.BeaconBlock
-
-	// BeaconState is a type alias for the BeaconState.
-	BeaconState = statedb.StateDB[
```
Comment (abi87) [inline cmd/beacond/types.go:148]: @fridrik01 I will miss the `BeaconState` name. Maybe we will re-introduce it at some point

File: beacon/blockchain/process_proposal.go [hunk 2]
```diff
@@ -181,6 +182,7 @@ func (s *Service[
 	// to avoid modifying the underlying state, for the event in which
 	// we have to rebuild a payload for this slot again, if we do not agree
 	// with the incoming block.
+	//nolint:contextcheck // TODO: We should look at using the passed context
```
Comment (abi87) [inline beacon/blockchain/process_proposal.go:185]: <3

**Review by abi87 — APPROVED** [review]
lgtm

**Review by calbera — APPROVED** [review]
lgtm

**Pull request merged**